### PR TITLE
docs: re-organize docs and introduce docs on concepts

### PR DESCRIPTION
**Pull request by hbagdi**
This is the first of a series of patches being made to add
thorough docs around Kong Ingress Controller.

This commit introduces the Concept docs, which provide users with
a high-level understanding of resources and deployment options.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,35 +1,66 @@
 # Kong Ingress Controller Documentation
 
-## Design
+## Table of content
```
Comment (levfishbluefish) [inline docs/README.md]:
```suggestion
## Table of contents
```

File: docs/concepts/custom-resources.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+Following CRDs allow users to declaratively configure all aspects of Kong:
+
+- [**KongIngress**](#kongingress)  
+- [**KongPlugin**](#kongplugin)
+- [**KongConsumer**](#kongconsumer)
+- [**KongCredential**](#kongcredential)
+
+## KongIngress
+
+Ingress resource in Kubernetes in a fairly narrow and ambiguous API, and
```
Comment (levfishbluefish) [inline docs/concepts/custom-resources.md]:
```suggestion
The Ingress resource in Kubernetes is a fairly narrow and ambiguous API, and
```

File: docs/concepts/custom-resources.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+It is not meant as a replacement to the `Ingress` resource in Kubernetes.
+Using `KongIngress`, all properties of [Upstream][kong-upstream],
+[Service][kong-service] and [Route][kong-route]
+entities in Kong related to an Ingress resource can be modified.
+
+Once a KongIngress resource is created, you can use `configuration.konghq.com`
+annotation to associate the KongIngress resource with an Ingress or a Service
+resource:
+
+- When the annotation is added too the Ingress resource, then the routing
```
Comment (levfishbluefish) [inline docs/concepts/custom-resources.md]:
```suggestion
- When the annotation is added to the Ingress resource, the routing
```

File: docs/concepts/custom-resources.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+## KongConsumer
+
+This custom resources configures Consumers in Kong.
+Every KongConsumer resource in Kubernetes directly translates to a
+[Consumer][kong-consumer] object in Kong.
+
+## KongCredential
+
+Once KongConsumer resource is created, credentials associated with the Consumer
```
Comment (levfishbluefish) [inline docs/concepts/custom-resources.md]:
```suggestion
Once a KongConsumer resource is created, credentials associated with the Consumer
```

File: docs/concepts/deployment.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
+# Kong Ingress Controller Deployment
+
+Kong Ingress Controller is designed to be deployed in a variety of ways
+to based on uses-cases. This document explains various components involved
```
Comment (levfishbluefish) [inline docs/concepts/deployment.md]:
```suggestion
based on uses-cases. This document explains various components involved
```

File: docs/concepts/deployment.md [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+and choices one can make as per the specific use-case.
+
+## Tables of content
```
Comment (levfishbluefish) [inline docs/concepts/deployment.md]:
```suggestion
## Table of Contents
```

File: docs/concepts/deployment.md [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+
+- [**Kubernetes Resources**](#kubernetes-resources):
+  Various Kubernetes resouces required to run Kong Ingress Controller.
+- [**Deployment options**](#deployment-options):
+  A high-level explanantion of choices that one should consider and customize
+  the deployment to best serve a specific use case.
+
+## Kubernetes Resources
+
+Following resources are used to run Kong Ingress Controller:
```
Comment (levfishbluefish) [inline docs/concepts/deployment.md]:
```suggestion
The following resources are used to run Kong Ingress Controller:
```

File: docs/concepts/deployment.md [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+Kong Ingress Controller can be deployed in any [namespace][k8s-namespace].  
+If Kong Ingress Controller is being used to proxy traffic for all namespaces
+in a Kubernetes cluster, which is generaly the case,
+it is recommended that it is installed in a dedicated
+`kong` namespace but it is not required to do so.
+
+The example deployments present in this repository automatically create a `kong`
+namespace and deploy resources into that namespace.
+
+### Custom resources
```
Comment (levfishbluefish) [inline docs/concepts/deployment.md]:
```suggestion
### Custom Resources
```

File: docs/concepts/deployment.md [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+
+> required
+
+Ingress resource in Kubernetes in a fairly narrow and ambiguous API, and
```
Comment (levfishbluefish) [inline docs/concepts/deployment.md]:
```suggestion
The Ingress resource in Kubernetes is a fairly narrow and ambiguous API, and
```

**Comment by hbagdi** [comment]
@levfishbluefish Thank you for the review, I've incorporated all of the changes. Please re-review.

**Pull request merged**